### DISCUSSION TITLE
Address change issue

### DISCUSSION THREAD

**Discussion by shaggione** [code4recovery/12-step-meeting-list#454] [Q&A] [answered]
When I attempt to change the address of a meeting (the meeting has changed location), I change the address:
![Image1](https://user-images.githubusercontent.com/86992860/125706856-820f355e-4db6-4f43-9326-ba539967eac2.png)
![Image2](https://user-images.githubusercontent.com/86992860/125706967-9fa1edc9-c54e-49a6-b78b-2a97475da033.png)
The map doesn't change...
I change the location:
![image](https://user-images.githubusercontent.com/86992860/125707133-0cc724e6-5632-4ca1-af15-a2a37e6b6e0a.png)
The map still doesn't change, and the publish button is greyed out. The save draft button is also greyed out.
![image](https://user-images.githubusercontent.com/86992860/125707262-0f5f54d7-b1dd-40b2-9914-c5d204187f26.png)
I have version 3.11.2 of the Meeting guide plug in and using Wordpress 5.7.2
My plug ins are:
12 Step Meeting List
Akismet Anti-Spam Version 4.1.10-Akismet Anti-Spam is automatically managed for you.
AMP Version 2.1.3
Classic Editor Version 1.6
CoBlocks Version 2.14.0-This plugin was installed by WordPress.com and provides features offered in your plan subscription.
Enable jQuery Migrate Helper Version 1.3.0
Gutenberg Version 11.0.0-This plugin was installed by WordPress.com and provides features offered in your plan subscription.
Jetpack Version 9.9-Jetpack is automatically managed for you.
Layout Grid Version 1.7-This plugin was installed by WordPress.com and provides features offered in your plan subscription.
Page Optimize Version 0.5.1-This plugin was installed by WordPress.com and provides features offered in your plan subscription.
PDF Embedder Version 4.6.2
WordPress.com Editing Toolkit Version 3.10777-This plugin was installed by WordPress.com and provides features offered in your plan subscription.
I tried disabling all plug ins (except for the Meeting Guide) and followed a sequence I found about changing address and the location. Nothing has worked.
Any suggestions?

**Comment by tech2serve** [comment]
Thank you for having tried disabling other plugins as that is a common source of issues. And I'm not sure yet we can rule that out in this case. The Gutenberg plugin has been known to cause issues with TSML.

Is this problem occurring with all address changes (or new meetings)? Or, just this one meeting?

I was able to geocode the address you provided on my dev site so I don't think we can point a finger at Google :-) What geocoder configuration are you using? If you're using the Legacy method, try our Beta API Gateway method. If you're using your own Google geocode key, check the settings for that through your Google account.

<img width="280" alt="image" src="https://user-images.githubusercontent.com/31904316/125834086-ea65f7b0-5fcb-4dd0-aab7-104f536157f7.png">

Please open up a dev/debugging console and try geocoding the address again. Please note any errors, and try to address them.

Another option is to go into another meeting and changes something *other* than the address, then click Update. This should rewrite the meeting list cache. That is a long shot imo, but worth trying.

**Reply by shaggione** [reply]
I opened the page and the address field (unchanged) was 4500 Buena Vista Rd, Bakersfield, CA 93311, USA:
Geocoded:  {formatted_address: "4500 Buena Vista Rd, Bakersfield, CA 93311, USA", latitude: 35.3135732, longitude: -119.1260703, approximate: "no", city: "Bakersfield", …}
When I changed the address to 10700 Brimhall Rd, Bakersfield, CA, USA, the code changed to:
Geocoded:  {status: "error", reason: "Google gave an unexpected response for address <co…y (\n  ),\n   'status' => 'REQUEST_DENIED',\n)</pre>"}
Some meetings allow changes, some do not.

**Reply by tech2serve** [reply]
What geocoder are you using? Legacy or our Beta API Gateway?

**Reply by shaggione** [reply]
I assume it's legacy as I was unaware of the Beta API Gateway.

**Reply by tech2serve** [reply]
Please switch to the Beta API Gateway, and try again. Please annotate the time you do the geocode, and I might be able to gather more information.

But, TBH, I'm leaning towards there being a conflict with one of your other plugins. If you would, please try disabling the Gutenberg plugin (and any others you're willing to try again with), clear your browser cache, and try the geocode again. Again, please note the times.

**Reply by shaggione** [reply]
I have no idea how to switch to the new API Gateway. I have disabled every plugin Wordpress doesn't control. Still cannot change a single meeting's address. If I change the location, the field is updated for all meetings at that address. Tried again on (or about 9:00 a.m. this morning) with no success.

**Reply by tech2serve** [reply]
You can select the API Gateway through the Import & Settings page, lower down on the right hand side.

Did you disable the Gutenberg plugin and try again?

**Comment by shaggione** [comment]
Disabled all plugins except:12 Step Meeting List
Akismet Anti-Spam (Automatically managed)
Jetpack (Automatically managed)
Very unfamiliar with WordPress (or web building in general). Just another servant!Import is in there somewhere...

    On Thursday, July 22, 2021, 09:08:35 PM PDT, Tech2Serve ***@***.***> wrote:  
 
 


You can select the API Gateway through the Import & Settings page, lower down on the right hand side.

Did you disable the Gutenberg plugin and try again?

**Reply by shaggione** [reply]
I deactivated all plugins not controlled by WordPress (including Gutenberg).

**Reply by tech2serve** [reply]
This is perplexing. I don't understand why Google is throwing the geocoding error of "REQUEST DENIED".

I may have missed your reply, but please confirm this is only happening to the one address? Or, are you unable to update any address?

The advantage of you switching to our API Gateway is that I can actually see the request being made to Google and can evaluate their response better. 

My gut is still telling me something is interfering with the contents of the request for this address. Since disabling Gutenberg didn't help, I'm left with a potential cache issue.

When you try to update the meeting, make sure to remove the contents of Location before entering the new Address. Also, you can try putting some other address and update the meeting, then try putting in the correct new address and updating? Again, please watch the errors in your console and record them.

Also, please contact your hosting platform (or log into your cPanel/account page) and try to clear the website cache.

Please report back with answers to any questions or suggestions I've made. It really helps me to help you.

**Comment by shaggione** [comment]
Even though I know little about WordPress and/or web design, I know enough to agree about the cache.Note: Address removed and I tabbed out of the field to refresh...
The map remains to the old address. Perplexing.I enter the new address (auto fill with the new address is shown):
The map doesn't change, the publish button is greyed out and the console reads as follows:
If I had any idea how to switch to your API Gateway, I would. I am skilled at desktop publishing, graphic design, 3d cad, 3d animation and play one HELL of a game of Solitaire! I quit advancing my web skills when anything but html was used 30 years ago. Just another railroaded servant. Thanks for what you are doing!As you requested, I changed the address to an existing location (the local Alano Club) and everything worked perfectly. Go figure...
(bad language)......What the heck? I will see if anyone knows who our hosting platform provider is (I think it's GoDaddy) and see if the site cache can be cleared.
    On Saturday, July 24, 2021, 08:41:56 AM PDT, Tech2Serve ***@***.***> wrote:  
 
 


This is perplexing. I don't understand why Google is throwing the geocoding error of "REQUEST DENIED".

I may have missed your reply, but please confirm this is only happening to the one address? Or, are you unable to update any address?

The advantage of you switching to our API Gateway is that I can actually see the request being made to Google and can evaluate their response better.

My gut is still telling me something is interfering with the contents of the request for this address. Since disabling Gutenberg didn't help, I'm left with a potential cache issue.

When you try to update the meeting, make sure to remove the contents of Location before entering the new Address. Also, you can try putting some other address and update the meeting, then try putting in the correct new address and updating? Again, please watch the errors in your console and record them.

Also, please contact your hosting platform (or log into your cPanel/account page) and try to clear the website cache.

Please report back with answers to any questions or suggestions I've made. It really helps me to help you.

**Reply by shaggione** [reply]
I contacted WordPress and the cache is cleared. Problem remains. I seems the pictures are not getting through. On the console, I saw this:
?v=2.0:16 WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: 
admin.min.js?ver=3.11.2:1 Geocoded:  
Object
reason: "Google gave an unexpected response for address <code>1015 Baker St, Bakersfield, CA 93305</code>. Response was <pre>(object) array(\n   'error_message' => 'This API project is not authorized to use this API.',\n   'results' => \n  array (\n  ),\n   'status' => 'REQUEST_DENIED',\n)</pre>"
status: "error"
__proto__:
constructor: ƒ Object()
hasOwnProperty: ƒ hasOwnProperty()
isPrototypeOf: ƒ isPrototypeOf()
propertyIsEnumerable: ƒ propertyIsEnumerable()
toLocaleString: ƒ toLocaleString()
toString: ƒ toString()
valueOf: ƒ valueOf()
__defineGetter__: ƒ __defineGetter__()
__defineSetter__: ƒ __defineSetter__()
__lookupGetter__: ƒ __lookupGetter__()
__lookupSetter__: ƒ __lookupSetter__()
get __proto__: ƒ __proto__()
set __proto__: ƒ __proto__()

**Reply by tech2serve** [reply]
So I'd like to take this convo offline and set up a time for us to do a Zoom call if you're agreeable with that. I'd like to look at your site remotely. I don't like to do this, but one of our members suggested that the problem might be with an entry in a cache we use in the plugin. Unfortunately, there is no way for you to clear that one.

I'm also noticing that you're now experiencing the issue with at least two different addresses?

If you're willing, please shoot me an email to devs@code4recovery.org.

And I know this is a bit of a nitpick, put please try to only reply as a comment to this thread (i.e., "write a reply" vice "Suggest an answer"). This will help make it easier to others to follow along.

**Reply by shaggione** [reply]
OK. I have no camera attached to my computer so a zoom call may be difficult. I can probably do it on my phone. I am having the problem on three different addresses. I understand about replying to the thread only. Still learning.

**Reply by tech2serve** [reply]
It is a limitation of GitHub Discussions. Ideally, I should be able to fold an "Answer" up into a comment like some other tools do. Not a biggie, just helps others a bit.

I've replied to your email. I do want to see the errors myself and help you that way.

**Reply by shaggione** [reply]
Team Viewer installed.

**Comment by tech2serve** [comment, marked as answer]
After discussion with the OP, we discovered his configuration for the geocoder had been changed to Google, but that his API key had not been authorized for geocoding.

Changing his configuration to use the API Gateway resolved his geocoding issues.
